src/Sylius/Component/Promotion/Generator/CouponGenerator.php

```diff
@@ -68,13 +72,19 @@ public function generate(PromotionInterface $promotion, Instruction $instruction
     /**
      * {@inheritdoc}
      */
-    public function generateUniqueCode()
+    public function generateUniqueCode($codeLength)
     {
         $code = null;
 
+        if (1 > $codeLength || 40 < $codeLength) {
```

Comment:
We use `sha1` to in this coupon generator so it is correct. It should not be in model as someone can impement generator that has different code length limitations.